drivers/hashmap/hashmap.go

```diff
@@ -167,12 +219,41 @@ func (db *Database) HealthCheck() error {
 		return hord.ErrNoDial
 	}
 
+	if db.config.Filename != "" {
+		_, err := os.Stat(db.config.Filename)
+		if err != nil {
+			return fmt.Errorf("error checking if file exists: %w", err)
+		}
+	}
+
 	return nil
 }
 
-// Close closes the hashmap database connection and clears all stored data.
+// Close closes the hashmap database connection and clears all stored data from memory (file remains if used).
 func (db *Database) Close() {
 	db.Lock()
 	defer db.Unlock()
 	db.data = nil
 }
+
+// saveToLocalFile is a helper function for methods that change the data (Set, Delete) and should
+// only be used after acquiring Write lock
+func (db *Database) saveToLocalFile() error {
+	if db.config.Filename == "" {
+		return nil
+	}
+
+	var err error
+	var content []byte
+	switch filepath.Ext(db.config.Filename) {
+	case ".json":
+		content, err = json.Marshal(db.data)
+	case ".yaml", ".yml":
+		content, err = yaml.Marshal(db.data)
+	}
+	if err != nil {
+		return err
+	}
+
+	return os.WriteFile(db.config.Filename, content, 0755)
```

Comment:
Same with this, if it returns error a bit more context around it would be useful